client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -179,9 +179,13 @@ const siteSetupFlow: Flow = {
 						);
 					}
 
-					// Add Launchpad to selected intents in General Onboarding
-					if ( isLaunchpadIntent( intent ) && typeof siteId === 'number' ) {
-						pendingActions.push( saveSiteSettings( siteId, { launchpad_screen: 'full' } ) );
+					// Update Launchpad option based on site intent
+					if ( siteId ) {
+						pendingActions.push(
+							saveSiteSettings( siteId, {
+								launchpad_screen: isLaunchpadIntent( intent ) ? 'full' : 'off',
+							} )
+						);
```

Comment:
This seems reasonable. Although Im wondering if there could be any issues setting it to `'off'` this early in the process (i know i recommended that in the issue but now having second thoughts).

IIRC we might use null vs 'off' as a distinguishing factor in some places like the my home stale data redirect logic? cc @jeyip.  If that is the case maybe setting it as null or empty string might make more sense. 🤔

Reply:
From what I can tell, using either value of null or ‘off’ would not affect cache invalidation and redirection in my home for the launchpad, but I’d vote for null or false as well

We only care about whether or not the value is full , and if it is, refetch data and make sure that the Launchpad wasn’t set to off recently

https://github.com/Automattic/wp-calypso/blob/e05b2fa84ddfd9671c51cb51f4aea3492228a71e/client/my-sites/customer-home/controller.jsx#L48-L50